lib/resources/aws/aws_ec2_instances.rb

```diff
@@ -0,0 +1,72 @@
+# author: Christoph Hartmann
+class AwsEc2Instances < Inspec.resource(1)
+  name 'aws_ec2_instances'
+  desc 'Verifies settings for an EC2 instance'
+  supports platform: 'aws'
+  example "
+    describe_ec2_instances.where { public_ip_address !=nil } do
+      its('instance_id') { should eq [] }
+    end
+  "
+
+  # TODO: rewrite to avoid direct injection, using AwsPluralResourceMixin
+  def initialize(opts = {}, conn = nil)
+    @opts = opts
+    @ec2_client = conn ? conn.ec2_client : inspec_runner.backend.aws_client(Aws::EC2::Client)
+  end
+
+  # Set up some accessors and the filter table
+  filter = FilterTable.create
+  filter.add_accessor(:where).add_accessor(:entries)
+  Aws::EC2::Types::Instance.members.each do |field|
+    # Add a filter table field for each AWS instance property
+    filter.add(field, field: field)
+    # Add an accessors to access properties without the filter table
+    define_method(field) do
+      list.map(&:field)
+    end
+  end
+  filter.connect(self, :list)
+
+  # Give uses of `inspec shell` a few hints about what they can test
+  def possible_properties
```

Comment:
Hmmm, I was not aware of this.  Great use of a feature!

Reply:
This is where I get the list of properties that need to be dynamically defined using `define_method`